networking/netinfo/netinfo.go

```diff
@@ -60,3 +71,24 @@ func Save(root string, info []NetInfo) error {
 
 	return json.NewEncoder(f).Encode(info)
 }
+
+// Merge the result of a CNI plugin's execution
+func (ni *NetInfo) MergeCNIResult(result types.Result) {
+
+	ni.IP = result.IP4.IP.IP
+	ni.Mask = net.IP(result.IP4.IP.Mask)
+	ni.HostIP = result.IP4.Gateway
+	ni.IP4 = result.IP4
+	ni.DNS = result.DNS
+}
+
+// the CNI spec passes around empty DNS configurations
+// instead of null pointers
+// This determines if the DNS struct has meaningful data
+func (ni *NetInfo) HasDNSConfig() bool {
+	d := ni.DNS
+	return !(len(d.Nameservers) == 0 &&
```

Comment:
This looks identical to `IsDNSZero` above. Can these two be unified or were they kept separated on purpose?